.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1004,6 +1004,28 @@ private DateTimeResolutionResult ParseOneWordPeriod(string text, DateObject refe
                         var date = referenceDate.AddYears(swift);
                         year = date.Year;
 
+                        if (!string.IsNullOrEmpty(match.Groups["special"].Value))
+                        {
+                            var specialYearTimex = this.config.SpecialYearPrefixesMap[match.Groups["special"].Value.ToLowerInvariant()];
```

Comment:
Timex generation should be in Timex-related classes. Pass to it the information and concepts it needs to decide how to create a timex. Don't pass internal parser state, ok?